### PR TITLE
Initial Ubuntu 18.04 Dockerfile with working msprime, and development…

### PR DESCRIPTION
**Pull request by agladstein**
… docs for Docker

Take a look at this Dockerfile. I played around with the dependencies, but this is what I got to work. Maybe there is a slimmer way of building the environment.
Also can consider starting from Alpine if we want the image to be smaller. Current image size is 817MB.

I'm testing the automated building on dockerhub:
https://cloud.docker.com/repository/docker/agladstein/test-auto-msprime

You can pull the image that was autobuilt:
```
docker pull agladstein/test-auto-msprime
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+            python3 \
+            python-dev \
+            python3-dev \
+            python-pip \
+            python3-pip \
+            build-essential \
+            libgsl0-dev \
+            libssl-dev \
```
Comment (jeromekelleher) [inline Dockerfile]: Do we need these libs below other than gsl0-dev? 

Also, I think we can use ``libgsl-dev`` these days

File: docs/development.rst [hunk 2]
```diff
@@ -481,6 +481,54 @@ QQ-plots of the statistics in question comparing ``ms`` and ``msprime``.
 There are also several "analytical" tests, which compare the distributions of
 values from ``msprime`` with analytical expectations.
 
+****************
+Containerization
+****************
+
+This repo is integrated with Dockerhub and the Docker image will be automatically
+built upon pushes to master on Dockerhub.
```
Comment (jeromekelleher) [inline docs/development.rst]: I don't think we should update the Docker image on updates to master. That would mean that we'd have to be very very careful about what we commit to master, and make the process of releasing a bit redundant. Can we instead update the Docker image when we tag a new release?

File: docs/development.rst [hunk 3]
```diff
@@ -481,6 +481,54 @@ QQ-plots of the statistics in question comparing ``ms`` and ``msprime``.
 There are also several "analytical" tests, which compare the distributions of
 values from ``msprime`` with analytical expectations.
 
+****************
+Containerization
+****************
+
+This repo is integrated with Dockerhub and the Docker image will be automatically
```
Comment (jeromekelleher) [inline docs/development.rst]: Maybe some links here? Perhaps a link to the Dockerhub page for ``tskit/msprime`` somewhere?

File: docs/development.rst (hunk 2 above)
Comment (agladstein) [reply docs/development.rst]: Yes, we can have it automatically update the Docker image with tagged releases and ignore master.

File: Dockerfile (hunk 1 above)
Comment (agladstein) [reply Dockerfile]: Just `libgsl-dev` worked. Thanks.

File: docs/development.rst [hunk 4]
```diff
@@ -481,6 +481,54 @@ QQ-plots of the statistics in question comparing ``ms`` and ``msprime``.
...
+Alternatively, msprime can be executed via the Docker container:
+.. code-block:: bash
+
+    $ sudo docker run -it test_msprime:latest mspms 10 1 -T
+
+Building Docker images and running Docker containers requires root access.
+If you are on a system and do not have root access, you can pull the Docker image
+and run it as a Singularity container.
+
+To run as a Singularity container, pull the docker image:
```
[the remark this answers is not in the document]
Comment (agladstein) [reply docs/development.rst]: Ok!

**Comment by jeromekelleher** [comment]
Merging; thanks @agladstein!

**Pull request merged**